src/test/java/com/google/devtools/build/lib/unix/UnixDigestHashAttributeNameTest.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.unix;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.devtools.build.lib.actions.cache.DigestUtils;
+import com.google.devtools.build.lib.vfs.DigestHashFunction;
+import com.google.devtools.build.lib.vfs.FileSystem;
+import com.google.devtools.build.lib.vfs.FileSystemTest;
+import com.google.devtools.build.lib.vfs.Path;
+import java.io.IOException;
+import org.junit.Test;
+
+/** Test for {@link com.google.devtools.build.lib.unix.UnixFileSystem#getFastDigest()}. */
+public class UnixDigestHashAttributeNameTest extends FileSystemTest {
```

Comment:
thanks for this! I'd also like a bit more of an end-to-end test. For instance, it's possible that somewhere Bazel is checking "can a fast hash possibly exist for this filesystem?", incorrectly getting no, and then digesting manually, especially since DigestUtils#manuallyComputeDigest is public. Would you mind adding such an integration test?

Reply:
I've just added a test along the lines that you described earlier on. I now run Bazel with `--profile=... --record_full_profiler_data` and check that the resulting log file contains `VFS xattr` entries.